downloader/src/test/kotlin/DownloaderTest.kt

```diff
@@ -0,0 +1,145 @@
+            val downloadedProjectDir = File(outputDir, pkg.normalizedName)
+            val downloadedModuleDir = File(downloadedProjectDir, submoduleName)
+
+            outputDir.list().size shouldBe 1
+            outputDir.list().first() shouldBe pkg.normalizedName
+
+            downloadedProjectDir.isDirectory shouldBe true
+            downloadedModuleDir.isDirectory shouldBe true
+
+            downloadedModuleDir.list().size shouldBe beGreaterThan(1)
+
+            val moduleDirList = downloadedModuleDir.list().asList()
+            moduleDirList should contain("build.gradle")
+            moduleDirList should contain("src")
+        }.config(tags = setOf(IntegrationTest))
+
+        "Downloads github oss-review-toolkit repo " {
+            val pkg = Package(
+                    packageManager = "Gradle",
+                    namespace = "",
+                    name = "oss-review-toolkit",
+                    version = "",
+                    declaredLicenses = sortedSetOf(),
+                    description = "",
+                    homepageUrl = "",
+                    downloadUrl = "",
+                    hash = "",
+                    hashAlgorithm = "",
+                    vcsPath = "",
+                    vcsProvider = "Git",
+                    vcsUrl = "git@github.com:heremaps/oss-review-toolkit.git",
+                    vcsRevision = ""
+            )
+            try {
+                Main.download(pkg, outputDir)
+            } catch (exception: DownloadException) {
+
+            }
+
+            val downloadDir = File(outputDir, pkg.normalizedName)
+            val downloadedPkgDirList = downloadDir.list().asList()
+            print("dir listing: $downloadedPkgDirList")
+            outputDir.list().size shouldBe 1
+            outputDir.list().first() shouldBe pkg.normalizedName
+            downloadedPkgDirList should contain("build.gradle")
+            downloadedPkgDirList should contain("downloader")
+            downloadedPkgDirList should contain("scanner")
+            downloadedPkgDirList should contain("model")
+            downloadedPkgDirList should contain("analyzer")
+            downloadedPkgDirList should contain("util")
+            downloadedPkgDirList should contain("graph")
+            downloadedPkgDirList should contain("docs")
+        }.config(tags = setOf(IntegrationTest))
+
+        "Downloads oss-review-toolkit repo zip from github" {
+            val pkg = Package(
+                    packageManager = "Gradle",
+                    namespace = "",
+                    name = "oss-review-toolkit",
+                    version = "",
+                    declaredLicenses = sortedSetOf(),
+                    description = "",
+                    homepageUrl = "",
+                    downloadUrl = "https://codeload.github.com/heremaps/oss-review-toolkit/zip/master",
+                    hash = "",
+                    hashAlgorithm = "",
+                    vcsPath = "",
```

Comment:
We meanwhile swapped arguments a bit, please make `vcsPath` go after `vcsRevision`.